### PR TITLE
A few small fixes

### PR DESCRIPTION
**Pull request by kpaschen**
I don't know if you'll want to merge all of these changes. I added checkpointing after every epoch because I had the trainer crash on one of the last epochs and ended up with no usable model file, but this might not fit into your workflows.

The key error typos should be fine to merge.

I also made the telegram messages optional and default to off. Personal preference.

* switch to checkpointing after every epoch; 
* fix key error in history (acc -> accuracy) and config (lr -> learning_rate)
* make telegram messages optional

### CODE REVIEW

**Comment by kpaschen** [comment]
sorry for the noise. i'll close this one and split it into two smaller requests.
